frontend/src/pages/UserView.js

```diff
@@ -83,6 +86,10 @@ function UserView() {
 			let currentField = updatedUserData;
 
 			for (let i = 0; i < pathArray.length - 1; i++) {
+				if(currentField[pathArray[i]] == undefined)
+				{
+					currentField[pathArray[i]] = {}; // add feild if it does not exist
```

Comment:
as per comment, added functionality to add field to data object if it doesn't exist.

This may actually the registration if you want to use this generic field code